File: service_booking_management/service_booking_management/report/booking_service_summary/booking_service_summary.py
# Copyright (c) 2025, Meghwin Dave and contributors
# For license information, please see license.txt

import frappe


def execute(filters=None):
	columns = [
		{"label": "Booking ID", "fieldname": "name", "fieldtype": "Link", "options": "Service Booking", "width": 120},
		{"label": "Customer Name", "fieldname": "customer_name", "fieldtype": "Data", "width": 180},
		{"label": "Service Type", "fieldname": "service_type", "fieldtype": "Data", "width": 120},
		{"label": "Preferred Datetime", "fieldname": "preferred_datetime", "fieldtype": "Datetime", "width": 160},
		{"label": "Status", "fieldname": "workflow_state", "fieldtype": "Data", "width": 100},
	]

	conditions = []
	values = {}
	if filters:
		if filters.get("service_type"):
			conditions.append("service_type = %(service_type)s")
			values["service_type"] = filters["service_type"]
		if filters.get("status"):
			conditions.append("workflow_state = %(status)s")
			values["status"] = filters["status"]

	where_clause = f"WHERE {' AND '.join(conditions)}" if conditions else ""

	data = frappe.db.sql(f"""
		SELECT
			name,
			customer_name,
			service_type,
			preferred_datetime,
			workflow_state
		FROM `tabService Booking`
		{where_clause}
		ORDER BY modified DESC
	""", values, as_dict=1)
	
	return columns, data

File: service_booking_management/service_booking_management/report/booking_service_summary/booking_service_summary.js
// Copyright (c) 2025, Meghwin Dave and contributors
// For license information, please see license.txt

frappe.query_reports["Booking Service Summary"] = {
	"filters": [
		{
			fieldname: "service_type",
			label: "Service Type",
			fieldtype: "Select",
			options: [
				"",
				"Therapy",
				"Spa",
				"Others"
			],
			default: ""
		},
		{
			fieldname: "status",
			label: "Status",
			fieldtype: "Select",
			options: [
				"",
				"Requested",
				"Approved",
				"Completed"
			],
			default: ""
		}
	],
	formatter: function(value, row, column, data, default_formatter) {
		if(column.fieldname === "workflow_state" && value) {
			let color = "gray";
			if(value === "Draft") color = "orange";
			else if(value === "Approved") color = "blue";
			else if(value === "Completed") color = "green";
			return `<span class=\"badge\" style=\"background:${color};color:white;padding:3px 10px;border-radius:8px;\">${value}</span>`;
		}
		return default_formatter(value, row, column, data);
	},
	get_chart_data: function(columns, results) {
		const counts = {};
		results.forEach(row => {
			const type = row[2]; // service_type
			counts[type] = (counts[type] || 0) + 1;
		});
		return {
			labels: Object.keys(counts),
			datasets: [
				{
					name: "Bookings",
					values: Object.values(counts)
				}
			]
		};
	},
	chart: {
		type: 'bar',
		height: 240
	}
};
